vowpalwabbit/autolink.h

```diff
@@ -1 +1,24 @@
+#pragma once
+
+#include "reductions.h"
+
 LEARNER::base_learner* autolink_setup(VW::config::options_i& options, vw& all);
+
+namespace VW
+{
+  struct autolink
+  {
+    autolink(uint32_t d, uint32_t stride_shift);
+    void predict(LEARNER::single_learner& base, example& ec);
+    void learn(LEARNER::single_learner& base, example& ec);
+
+  private:
+    void prepare_example(LEARNER::single_learner& base, example& ec);
+    void reset_example(example& ec);
+
+    // degree of the polynomial
+    const uint32_t m_d;
+    const uint32_t m_stride_shift;
```

Comment:
I have been using _ in most of the cpp code instead of m_.  I like it because it's one letter instead of two.

Reply:
We should have a quick discussion to figure out all our naming conventions. Namespaces, object names, variable names, etc

Reply:
In the past I've used m_, s_ and c_ to indicate member, static and constant respectively.